controllers/osbuildenvconfig_controller_test.go

```diff
@@ -961,25 +1006,639 @@ var _ = Describe("OSBuildEnvConfig Controller", func() {
 																// when
 																result, err := reconciler.Reconcile(requestContext, request)
 																// then
-																Expect(err).To(BeNil())
-																Expect(result).To(Equal(resultDone))
+																Expect(err).To(Equal(errFailed))
+																Expect(result).To(Equal(resultRequeue))
+															})
+
+															Context("Secret for osbuild-worker ssh keys not found", func() {
+																BeforeEach(func() {
+																	secretRepository.EXPECT().Read(requestContext, workerSSHKeysSecretName, operatorNamespace).Return(nil, errNotFound)
+																})
+
+																It("Should return en error if failed to create the ssh keys", func() {
+																	// given
+																	sshGenerateError := fmt.Errorf("failed to generate ssh keys")
+																	sshkeyGenerator.EXPECT().GenerateSSHKeyPair().Return(nil, nil, sshGenerateError)
+																	// when
+																	result, err := reconciler.Reconcile(requestContext, request)
+																	// then
+																	Expect(err).To(Equal(sshGenerateError))
+																	Expect(result).To(Equal(resultRequeue))
+																})
+
+																Context("Successfully created the ssh keys", func() {
+																	BeforeEach(func() {
+																		sshkeyGenerator.EXPECT().GenerateSSHKeyPair().Return(sshPrivateKey, sshPublicKey, nil)
+																	})
+
+																	It("Should return an error if failed to create the secret for the osbuild-worker ssh keys", func() {
+																		// given
+																		secretRepository.EXPECT().Create(requestContext, workerSSHKeysSecret).Return(errFailed)
+																		// when
+																		result, err := reconciler.Reconcile(requestContext, request)
+																		// then
+																		Expect(err).To(Equal(errFailed))
+																		Expect(result).To(Equal(resultRequeue))
+																	})
+
+																	It("Should return requeue if succeeded to create the secret for the osbuild-worker ssh keys", func() {
+																		// given
+																		secretRepository.EXPECT().Create(requestContext, workerSSHKeysSecret).Return(nil)
+																		// when
+																		result, err := reconciler.Reconcile(requestContext, request)
+																		// then
+																		Expect(err).To(BeNil())
+																		Expect(result).To(Equal(resultQuickRequeue))
+																	})
+																})
+															})
+
+															Context("Secret for osbuild-worker ssh keys exists", func() {
+																BeforeEach(func() {
+																	secretRepository.EXPECT().Read(requestContext, workerSSHKeysSecretName, operatorNamespace).Return(workerSSHKeysSecret, nil)
+																})
+
+																It("Should return an error if failed to get the virtualmachine for the internal builder", func() {
```

Comment:
virtualmachin -> VirtualMachine  (just because I saw in other places you used the upper case for the description )

Reply:
Done